testing/sdk_networks.py

```diff
@@ -19,3 +27,23 @@ def check_task_network(task_name, expected_network_name="dcos"):
                     assert "name" not in network_info, \
                         "Task {task} has network name when it shouldn't has status:{status}" \
                         .format(task=task_name, status=status)
+
+
+def get_endpoints(endpoint_to_get, package_name, correct_count):
+    endpoints, _, rc = shakedown.run_dcos_command("{} endpoints {}".format(package_name, endpoint_to_get))
+    assert rc == 0, "Failed to get endpoints on overlay network"
+    endpoints = json.loads(endpoints)
+    assert len(endpoints) == correct_count, "Wrong number of endpoints, got {} should be {}" \
+        .format(len(endpoints), correct_count)
+    return endpoints
+
+
+def check_endpoints_on_overlay(endpoints):
+    ip_addresses = [e.split(":")[0] for e in endpoints["address"]]
+    assert len(set(ip_addresses).intersection(set(shakedown.get_agents()))) == 0, \
+        "IP addresses for this service should not contain agent IPs, IPs were {}".format(ip_addresses)
+
+    assert "dns" in endpoints, "Endpoints missing DNS key"
+    for dns in endpoints["dns"]:
+        assert "autoip.dcos.thisdcos.directory" in dns, "DNS {} is incorrect should have autoip.dcos.thisdcos." \
```

Comment:
Not quite, the DNS will be `<task_name>.<service_name>.autoip.dcos.thisdcos.directory:<port>`, `endswith()` requires that I know the port.